src/System.Collections/src/System/Collections/BitArray.cs

```diff
@@ -398,17 +398,49 @@ public void CopyTo(Array array, int index)
 
             if (array is int[])
             {
-                Array.Copy(m_array, 0, array, index, GetArrayLength(m_length, BitsPerInt32));
+                int last = GetArrayLength(m_length, BitsPerInt32) - 1;
+                int bits = m_length % BitsPerInt32;
+
+                if (bits == 0)
+                {
+                    // we have perfect bit alignment, no need to sanitize, just copy
+                    Array.Copy(m_array, 0, array, index, GetArrayLength(m_length, BitsPerInt32));
+                }
+                else
+                {
+                    // do not copy the last int, as it is not completely used
+                    Array.Copy(m_array, 0, array, index, GetArrayLength(m_length, BitsPerInt32) - 1);
+
+                    // the last int needs to be masked
+                    ((int[])array)[last] = m_array[last] & (1 << bits) - 1;
```

Comment:
Also consider changing the if check earlier to be:
```C#
int[] intArray = array as int[];
if (intArray != null)
```
so that here this can just be:
```C#
intArray[last] = m_array[last] & ((1 << bits) - 1);
```